ambry-cloud/src/main/java/com/github/ambry/cloud/HelixVcrClusterParticipant.java

```diff
@@ -195,7 +195,8 @@ public void participate() throws Exception {
     helixAdmin = manager.getClusterManagmentTool();
 
     InstanceConfig instanceConfig = helixAdmin.getInstanceConfig(vcrClusterName, vcrInstanceName);
-    if (!instanceConfig.getRecord().getBooleanField(VCR_HELIX_CONFIG_READY, false)) {
+    if (!instanceConfig.getRecord().getBooleanField(VCR_HELIX_CONFIG_READY, false) ||
```

Comment:
is there any cases where vcrHelixConfigReady is false? I mean, I don't see HelixVcrClusterParticipant ever update this to false.

Reply:
I didn't see any case either.

Reply:
i think in this case, we should not use any boolean value to indicate if we should should update the instance config. I think we should define an VcrInstanceConfig class and compare the object we obtain from helix and the object we expect. If they are not the same, we update the helix.

Reply:
I've added the VcrInstanceConfig
1. We'd better use VcrInstanceConfig in HelixVcrClusterParticipant and HelixVcrClusterSpectator and other related vcr code. It needs a code refactoring. It's out of the scope of the PR and won't address in this PR.
2. I'd like to still keep the VCR_HELIX_CONFIG_READY. Although in our code, we don't modify it or set it to false anywhere. But it's a flag we can use to change the vcr using some other Operational tooling. So I prefer keeping it.

Reply:
Filed new revision. Added VCR_HELIX_CONFIG_READY to the VcrInstanceConfig.